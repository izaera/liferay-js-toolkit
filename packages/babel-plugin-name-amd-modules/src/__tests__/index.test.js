import * as babel from 'babel-core';
import plugin from '../index';

it('correctly names anonymous modules', () => {
	const source = `
	define([], function(){})
	`;

	const { code } = babel.transform(source, {
		filenameRelative: __filename,
		plugins: [plugin],
	});

	expect(code).toMatchSnapshot();
});

it('correctly renames named modules', () => {
	const source = `
	define('my-invalid-name', [], function(){})
	`;

	const { code } = babel.transform(source, {
		filenameRelative: __filename,
		plugins: [plugin],
	});

	expect(code).toMatchSnapshot();
});
