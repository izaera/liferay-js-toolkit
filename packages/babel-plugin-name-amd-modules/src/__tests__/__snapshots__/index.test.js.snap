// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly names anonymous modules 1`] = `
"
define(\\"package@1.0.0/index.test\\", [], function () {});"
`;

exports[`correctly renames named modules 1`] = `
"
define('package@1.0.0/index.test', [], function () {});"
`;
